src/gen.rs

```diff
@@ -132,14 +132,34 @@ impl Generator {
     }
 }
 
+/// Utility function to find nested float in record and it's dependencies.
+fn deep_search_float(schema: &Schema, deps_map: &HashMap<Name, Schema>) -> Result<bool> {
+    match schema {
+        Schema::Float | Schema::Double => Ok(true),
+        Schema::Array(inner) | Schema::Map(inner) => deep_search_float(inner, deps_map),
+        Schema::Record { fields, .. } => Ok(fields
+            .iter()
+            .any(|f| deep_search_float(&f.schema, deps_map).unwrap())),
+        Schema::Union(union) => Ok(union
+            .variants()
+            .iter()
+            .any(|s| deep_search_float(s, deps_map).unwrap())),
+        Schema::Ref { name } => deps_map
+            .get(name)
+            .ok_or_else(|| Error::Schema("Not found".to_owned()))
```

Comment:
I find it very useful for debugging purposes to have the schema `name` in the error message.

Reply:
Done